internal/lock/lockfile.go

```diff
@@ -127,6 +128,10 @@ func IsLegacyPackage(pkg string) bool {
 		!strings.HasPrefix(pkg, "/")
 }
 
+func (l *File) Tidy(project devboxProject) {
+	l.PackageMap = lo.PickByKeys(l.PackageMap, project.Packages())
```

Comment:
@mikeland86 not 100% confident that the keys in the lockfile are _intended_ to match the package-names in devbox.json. Can you confirm?

Reply:
Yeah they should always match. Right now `nix.Input` uses `Input.String()` instead of `Input.Raw` as a key (raw is the package name in devbox.json) but they always match for devbox packages. I think we should probably change those callsites to use `input.Raw` just to be sure (since for flakes `Raw` and `String()` might not be the same)

But for this use case this is safe.

Reply:
oh I see. We should probably change that sooner than later...